lib/url-pattern/url-match-pattern.js

```diff
@@ -64,19 +80,27 @@ _.assign(UrlMatchPattern.prototype, /** @lends UrlMatchPattern.prototype */ {
      */
     createMatchPattern: function () {
         var matchPattern = this.pattern,
+            allowProtocols = _.isArray(this.protocols),
```

Comment:
Instead of all these, why not give a getter function `.protocol` which returns the protocol from the match itself?